.buildkite/pipeline.elastic-agent-binary-dra.yml

```diff
@@ -1,3 +1,34 @@
 steps:
-  - label: "Example test"
-    command: echo "Hello!"
+  - group: ":beats: DRA Elastic-Agent Core Snapshot :beats:"
+    key: "dra-core-snapshot"
+    steps:
+    - label: ":hammer::package: Build and DRA Publish Elastic-Agent Core Snapshot"
+      if: build.branch == 'main' || build.branch =~ /^[0-9]+\.[0-9]+\$/ || build.env("RUN_SNAPSHOT") == "true"
+      commands:
+        - .buildkite/scripts/steps/dra-build.sh
+        - .buildkite/scripts/steps/dra-publish.sh
+      key: "build-and-publish-dra-snapshot"
+      agents:
+        provider: "gcp"
+        machineType: "c2-standard-16"
+      env:
+        WORKFLOW: "snapshot"
+  - group: ":beats: DRA Elastic-Agent Core Staging :beats:"
+    key: "dra-core-staging"
+    steps:
+    - label: ":hammer::package:  Build and DRA Publish Elastic-Agent Core staging"
+      if: build.branch =~ /^[0-9]+\.[0-9]+\$/ || build.env("RUN_STAGING") == "true"
+      commands:
+        - .buildkite/scripts/steps/dra-build.sh
+        - .buildkite/scripts/steps/dra-publish.sh
```

Comment:
when triaging issues, having the publish be as a separate step would help quickly understand where the failure is.
I would advice having the `dra-publish` as a separate step in the pipelines, so each step has one role in the process.